### PR TITLE
Added LV Sense SoC to CAN

### LINKED ISSUE #219: [Sense] - Create Accurate LV SoC Calcs
### Description

Create and send LV SoC calcs along with the voltage itself, in the same CAN message.

### Acceptance Criteria

SoC calcs are accurate to the discharge curve and relatively linear.

### Proposed Solution

Use the datasheet for the 24A LV battery, edit the CAN message spec in JSON.

### PR DESCRIPTION
**Pull request by harrison-e**
## Changes

- Added LV Sense SoC calculation using logistic function to `Core/Src/monitor.c`
- Added `#include <math.h>` to `Core/Src/monitor.c`
- Changed CAN spec in `mpu.json` to include SOC NetField in `0x503` (See https://github.com/Northeastern-Electric-Racing/Embedded-Base/pull/212)

Closes #219

### CODE REVIEW

**Comment by harrison-e** [comment]
Note that a better version of this uses Coulomb counting!

File: Core/Src/monitor.c [hunk 1]
```diff
@@ -51,7 +58,26 @@ void read_lv_sense(void *arg)
...
+	// Calculate SoC using logistic function
+	// - Normal charged voltage is 29.4V
+	// - 18650 max charged voltage is 4.2V
+	// - 29.4V/4.2V = 7 batteries
+	// - Divide v_dec by 7 to get avg voltage over all rows
+	// - Use logistic function to model SoC
+	// https://www.aegisbattery.com/products/24v-20ah-li-ion-battery-pvc
+	float v_max = 4.2; // max avg voltage over all rows (7)
+	float v_min = 2.8; // min avg voltage over all rows (7)
+	float k = -8.5; // logistic fn parameter to affect steepness of curve
```
Comment (jr1221) [inline Core/Src/monitor.c]: label parts of algo `static const` if they do not change for readability sake, altough the compiler will optimize this stuff away anyways.

File: Core/Src/monitor.c [hunk 2]
```diff
@@ -51,7 +58,26 @@ void read_lv_sense(void *arg)
...
+	float i = (v_max + v_min) /
+		  2; // logistic fn parameter to affect midpoint of curve
+	float soc_dec =
+		1 / (1 + exp(k * (v_dec - i))); // SoC calculation in [0,1]
+	soc_int = soc_dec * 100;
```
Comment (jr1221) [inline Core/Src/monitor.c:75]: so we apply a 100 scale to this 0 to 1 value, so the data sent over CAN is casted 0 to 100.  We can do to things to clarify this:
- Add a prescaler and corresponding JSON postscaler which allows us to get decimal points out of our SoC
- Or: Use a uint8_t instead as this is a number 0 to 100 so mine as well save space.

**Review by jr1221 — CHANGES_REQUESTED** [review]
see question about soc scale over CAN.

**Review by jr1221 — APPROVED** [review]
Lgtm.  Obv math may not work but the framework is there so that's good.

**Pull request merged**